src/hooks/useRequest.test.ts

```diff
@@ -0,0 +1,31 @@
+import { renderHook } from 'react-hooks-testing-library';
+import { queryGql } from '../test-utils';
+import { useRequest } from './useRequest';
+
+it('preserves instance of request when key has not changed', () => {
+  let { query, variables } = queryGql;
+
+  const { result, rerender } = renderHook(
+    ({ query, variables }) => useRequest(query, variables),
```

Comment:
I think we might be unintentionally shadowing `query` and `variables` here. Typically I'll define what's in `initialProps` to be different from my test data to ensure that all calls to `rerender` are using the data passed in `initialProps` or `rerender`, i.e.:

```
const { result, rerender } = renderHook(
   ({ q, v }) => useRequest(q, v),
   { initialProps: { q: query, v: variables }
);
```

Functionally it doesn't matter b/c `query` (inside `renderHook`) -> `initialProps.query` -> `query` (from `queryGql`), and we use `let` bindings to update the references to `variables` below, so this is more of a readability suggestion to make it clear where things are coming from 🤗